[Hub Generated] Review request for Microsoft.Migrate/AssessmentProjects to add version stable/2023-03-15

kuchatur-ms opened the pull request:
This is a PR generated at [OpenAPI Hub](https://aka.ms/openapihub). You can view your [work branch via this link](https://openapihub.azure-devex-tools.com/branch/Azure/azure-rest-api-specs/kuchatur-ms-migrate-Microsoft.Migrate-2023-03-15-test1...main/).
# ARM (Control Plane) API Specification Update Pull Request 

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

![diagram](https://github.com/Azure/azure-rest-api-specs/assets/4429827/b321e1dc-5e07-4940-9cd6-a65ac6729a8b)

[1] [ARM review queue] (for **merge** queues, see [4])  
The PRs are processed by time opened, ascending. Your PR may show up on 2nd or later page. 
If you addressed Step 1 from the diagram and your PR is not showing up in the queue, ensure the label `ARMChangesRequested` 
is removed from your PR. This should cause the label `WaitForARMFeedback` to be added.  
[2] https://aka.ms/azsdk/support/specreview-channel  
[3] [List of SDK breaking changes approvers] in pinned Teams announcement  
[4] [public repo merge queue], [private repo merge queue] (for **ARM review** queue, [1])

If you need further help with anything, see `Getting help` section below.

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [x] New API version. (Such PR should have been generated with [OpenAPI Hub](https://aka.ms/openapihub), per [this wiki doc](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).)
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to Step 2, "ARM Review", for this PR.

### Breaking changes review (Step 1)

- If the automation determines you have breaking changes, i.e. Step 1 from the diagram applies to you,
  you must follow the [breaking changes process].  
  **IMPORTANT** This applies even if:
  - The tool fails while it shouldn't, e.g. due to runtime exception, or incorrect detection of breaking changes.
  - You believe there is no need for you to request breaking change approval, for any reason. 
    Such claims must be reviewed, and the process is the same.

### ARM API changes review (Step 2)

- If this PR is in purview of ARM review then automation will add the `ARMReview` label.
- If you want to force ARM review, add the label yourself.
- Proceed according to the diagram at the top of this comment.

## Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

## Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/85/Swagger-Suppression-Process) 
to get approval.

## Getting help

- For general PR approval workflow, see the diagram at the top of this comment.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[ARM review queue]: https://github.com/search?q=org%3AAzure+is%3Apr+is%3Aopen+label%3AWaitForARMFeedback+-label%3AIDCDevDiv++draft%3Afalse+sort%3Acreated-asc+&type=pullrequests
[List of SDK breaking changes approvers]: https://teams.microsoft.com/l/message/19:0351f5f9404446e4b4fd4eaf2c27448d@thread.skype/1689115217750?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1689115217750&teamName=Azure%20SDK&channelName=API%20Spec%20Review&createdTime=1689115217750
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AApproved-OkToMerge+draft%3Afalse+sort%3Acreated-asc
[breaking changes process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

kuchatur-ms committed:
Adds base for updating Microsoft.Migrate/AssessmentProjects from version stable/2019-10-01 to version 2023-03-15

kuchatur-ms committed:
Updates readme

kuchatur-ms committed:
Updates API version in new specs and examples

kuchatur-ms committed:
edit

kuchatur-ms committed:
added types files

kuchatur-ms committed:
added examples

kuchatur-ms committed:
edit. spelling ignore words added

kuchatur-ms committed:
edit. api version fix

kuchatur-ms committed:
edit.

kuchatur-ms committed:
edit.

kuchatur-ms committed:
edit.

kuchatur-ms committed:
deleted zombie files and ran prettier

kuchatur-ms committed:
removed data plane folder to fix checks

kuchatur-ms committed:
Merge remote-tracking branch 'origin/main' into kuchatur-ms-migrate-Microsoft.Migrate-2023-03-15-test1

sjanamma commented on specification/migrate/common-types/resource-management/v1/privatelinks.json:
```diff
@@ -0,0 +1,182 @@
+{
```
can you use the common-types instead of creating another version?

kuchatur-ms replied:
the swagger generated from typespec requires the common types to be in a certain directory relative to the current one. I'll check with what changes we took for SDS PR and update.

kuchatur-ms replied:
ive removed it. Now there is an error in lint diff. Should i just ignore it?

kuchatur-ms replied:
fixed this.

kuchatur-ms committed:
Removed common types folder.

kuchatur-ms committed:
Merge remote-tracking branch 'origin/main' into kuchatur-ms-migrate-Microsoft.Migrate-2023-03-15-test1

kuchatur-ms committed:
Fixed relative path to common types + minor swagger edits

kuchatur-ms committed:
removed 2019 from default tag to fix go sdk check

kuchatur-ms committed:
edit

kuchatur-ms committed:
minor swagger fixes

kuchatur-ms committed:
edit

kuchatur-ms committed:
Merge remote-tracking branch 'origin/main' into kuchatur-ms-migrate-Microsoft.Migrate-2023-03-15-test1

TimLovellSmith commented on cSpell.json:
```diff
@@ -402,9 +402,21 @@
+                "Overcommit",
+                "overcommit",
+                "SSDE",
+                "Recieved",
```
>Recieved [](http://example.com/codeflow?start=17&length=8)

this is a misspelling, is it in a context where it can't be fixed, like an enum?

TimLovellSmith replied:
looks like it ;(

TimLovellSmith replied:
I don't see it in the old (Baseline) api though which suggests it is a new API - and we can fix the spelling?

kuchatur-ms replied:
We have fixed any spelling mistake we could fix in our code. These ones are spelling mistakes in enums/ properties which have already been shipped. Now we cant change these properties as it would break the customers.

We have had API versions released between the last published one and this one, but we discussed with the ARM team/ breaking change team to release 1 clean API version and deprecate the old ones without releasing the swagger for those. Thats why you might not see these properties in old API versions, but sadly they have already been released a while ago

TimLovellSmith replied:
Were those API versions preview, or GA versions? @kuchatur-ms

kuchatur-ms replied:
There were both old GA and preview API versions.

TimLovellSmith commented:
Please look into the policheck and modelvalidation failures.

Please also replace the autogenerated examples with realistic ones, to aid in review and API validation

TimLovellSmith commented on cSpell.json:
```diff
@@ -402,9 +402,21 @@
+                "Ebds",
+                "idms",
+                "Idms",
+                "Cluter",
```
>Cluter [](http://example.com/codeflow?start=17&length=6)

Can we remove it?

kuchatur-ms replied:
We have fixed any spelling mistake we could fix in our code. These ones are spelling mistakes in enums/ properties which have already been shipped. Now we cant change these properties as it would break the customers.

TimLovellSmith commented on cSpell.json:
```diff
@@ -402,9 +402,21 @@
+            "filename": "**/specification/migrate/resource-manager/**/*.json",
             "words": [
-                "Occured"
+                "Occured",
```
>Occured [](http://example.com/codeflow?start=17&length=7)

Can we fix the misspelling? 'Occured-> Occurred' I am imagining

kuchatur-ms replied:
We have fixed any spelling mistake we could fix in our code. These ones are spelling mistakes in enums/ properties which have already been shipped. Now we cant change these properties as it would break the customers.

kuchatur-ms commented:
> Please look into the policheck and modelvalidation failures.
> 
> Please also replace the autogenerated examples with realistic ones, to aid in review and API validation

The model validation failures check if failing incorrectly as the examples were generated from the oav examples generator. We pushed another PR recently and had asked the reviewer to bypass this check. ref: https://github.com/Azure/azure-rest-api-specs/pull/25366/checks?check_run_id=16521415969
Can you do the same for this PR?

PoliCheck if failing because looks like the autogenerated example had the word "pee" in one of the strings XD. I'll update that.

The examples are generated using the supplied oav example generator tool. Manually generating examples for 100+ APIs is not feasible. If the PR team has a generic tool which can generate the same for us using our swagger, we can use that to generate realistic examples.

kuchatur-ms committed:
edited string for policheck

TimLovellSmith commented:
>anually generating examples for 100+ APIs is not feasible

I have a feeling 'find and replace' might be an easy way to make some quick improvements.

kuchatur-ms commented:
> > manually generating examples for 100+ APIs is not feasible
> 
> I have a feeling 'find and replace' might be an easy way to make some quick improvements.

We can do that for properties which are common in multiple resources, but most resources will have unique properites which will have to be updated manually. Also if we take a fix in this API version later and use the example generator tool, it will override this manual change.
Note that we will be raising another couple of PRs for this RP very soon.
ALso, I've seen most services use auto-generated examples and this is what is also suggested in the documents. As i mentioned before, the only feasible way to do this long term is to use a tool which can make API calls on actual resources and store those results in examples files. I know of a similar tool written by the powershell team which works for generating realistic examples for powershell commandlets. We can take help of any such tool in future, but for now, we'd like to avoid taking manual changes in example files.

kuchatur-ms committed:
minor fix

kuchatur-ms committed:
Added manual examples

kuchatur-ms committed:
Merge remote-tracking branch 'origin/main' into kuchatur-ms-migrate-Microsoft.Migrate-2023-03-15-test1

kuchatur-ms commented:
Manual examples are now added for all APIs

kuchatur-ms committed:
fixed errors in checks

raosuhas commented:
Can you please explain the diff between the 2019-10-01 api version to this one ?  Why did you decide to completely restructure this file ? This change makes it so that we have no way to see the diff so would end up having to review the entire file now. 

Also you seem to have a large number of avocado failures. Can you please check and fix those as well ?

kuchatur-ms committed:
testing update readme to latest API version

kuchatur-ms commented:
> Can you please explain the diff between the 2019-10-01 api version to this one ? Why did you decide to completely restructure this file ? This change makes it so that we have no way to see the diff so would end up having to review the entire file now.
> 
> Also you seem to have a large number of avocado failures. Can you please check and fix those as well ?

We had published the swagger for the 2019 API version a long time back. There have been lots of feature additions and changes since then, hence the large diff. Also the old API wasnt following ARM guidelines properly, hence we had to take some breaking changes to fix those issues.
Rather than maintianing the swagger manually, we have also started using typespec according to API authoring guidelines from ARM team, which is why the swagger generated is restructered. These changes are required and i see no other way to clean things up and follow the guidelines without taking them. We hope in future our APIs are cleaner and we will have minimal changes in these files.
For swagger avocado, I have fixed some errors, but some will stay as we have taken breaking changes in the API. For a similar PR we took very recently we were able to get a hallpass for this check. https://github.com/Azure/azure-rest-api-specs/pull/25366/checks?check_run_id=16521415499
As the breaking changes are approved already, we can approve the PR without the avocado checks for now.

kuchatur-ms commented:
> > Can you please explain the diff between the 2019-10-01 api version to this one ? Why did you decide to completely restructure this file ? This change makes it so that we have no way to see the diff so would end up having to review the entire file now.
> > Also you seem to have a large number of avocado failures. Can you please check and fix those as well ?
> 
> We had published the swagger for the 2019 API version a long time back. There have been lots of feature additions and changes since then, hence the large diff. Also the old API wasnt following ARM guidelines properly, hence we had to take some breaking changes to fix those issues. Rather than maintianing the swagger manually, we have also started using typespec according to API authoring guidelines from ARM team, which is why the swagger generated is restructered. These changes are required and i see no other way to clean things up and follow the guidelines without taking them. We hope in future our APIs are cleaner and we will have minimal changes in these files. For swagger avocado, I have fixed some errors, but some will stay as we have taken breaking changes in the API. For a similar PR we took very recently we were able to get a hallpass for this check. https://github.com/Azure/azure-rest-api-specs/pull/25366/checks?check_run_id=16521415499 As the breaking changes are approved already, we can approve the PR without the avocado checks for now.

Also note that we have not published any SDK for any API version till now. We plan to publish our first SDKs in December time frame so the SDK related issues will be worked on in future.

kuchatur-ms commented:
After updating the API version in readme these are the merging requirements results
❌ ARMSignedOff
❌ Approved-SdkBreakingChange-Go
❌ Approved-SdkBreakingChange-JavaScript
The last two are releated to SDK breaking changes which dont apply to us as we dont publish any SDK right now. The first SDK will be published in December. As we already have signoff on the breaking changes in the API, and we dont have any breaking changes in the SDK, can you bypass these checks and approve the PR?

Otherwise i can revert the commit where i changed the API version, we can ignore the avocado checks and merge the PR. When we publish SDK in December, then we'll take these changes again.

krdhruva commented:
>             "headers": {

How would client retrieve operation status? Consider using location header for URL query.

---
Refers to: specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/stable/2023-03-15/migrate.json:323 in cdb7301. [](commit_id = cdb7301bd381a52875e3b0c98b39f501e495faf2, deletion_comment = False)

krdhruva commented:
>               "Retry-After": {

Consider adding location header to return url to track long running operation.

---
Refers to: specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/stable/2023-03-15/migrate.json:324 in cdb7301. [](commit_id = cdb7301bd381a52875e3b0c98b39f501e495faf2, deletion_comment = False)

krdhruva commented on specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/stable/2023-03-15/examples/AssessmentProjectsOperations_Create_MaximumSet_Gen.json:
```diff
@@ -0,0 +1,144 @@
+      }
+    },
+    "201": {
+      "headers": {
```
"Retry-After" header is missing in example.

kuchatur-ms replied:
Our services dont send retry-after header. As per the API reference, retry header is an optional thing. While we will continue to improve our APIs in future and will work to take this up later, we will not be taking this now.
https://github.com/cloud-and-ai-microsoft/resource-provider-contract/blob/master/v1.0/async-api-reference.md#202-accepted-and-location-headers

krdhruva commented:
>             "name": "hyperCollectorName",

hypervCollectorName?

---
Refers to: specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/stable/2023-03-15/migrate.json:3280 in cdb7301. [](commit_id = cdb7301bd381a52875e3b0c98b39f501e495faf2, deletion_comment = False)

krdhruva commented:
>             "name": "hyperCollectorName",

hypervCollectorName?

---
Refers to: specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/stable/2023-03-15/migrate.json:3331 in cdb7301. [](commit_id = cdb7301bd381a52875e3b0c98b39f501e495faf2, deletion_comment = False)

kuchatur-ms committed:
edit

kuchatur-ms committed:
edit

kuchatur-ms commented:
Looks like there have been recent changes to the lint diff check which are now flagging the changes with some errors. These checks were passing before:

1. Before: https://github.com/Azure/azure-rest-api-specs/pull/25417/checks?check_run_id=17205671065
2. Now: https://github.com/Azure/azure-rest-api-specs/pull/25417/checks?check_run_id=17340779929

The errors here are of the following types, and we cant take these changes for these reasons:
1. pattern for resource name property: These APIs expose new API version for existing APIs. These APIs were released without pattern checks and now we cant add a pattern to match here while ensuring the API has no regressions for existing resources.
2. Long running operations should return xyz status code: These checks werent failing till Friday. We can take these changes in a fix PR in the near future. For now, we can continue with this PR as it was passing checks till now.

krdhruva commented:
>     "PrivateLinkResourceListResult": {

Consider using private link properties from common properties:
[https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/privatelinks.json](https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/privatelinks.json)

---
Refers to: specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/stable/2023-03-15/migrate.json:12873 in 7efa027. [](commit_id = 7efa0272dc9ada11abe239c90e8b202f5d6f2a67, deletion_comment = False)

TimLovellSmith commented:
> > ```
> > "PrivateLinkResourceListResult": {
> > ```
> 
> Consider using private link properties from common properties: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/privatelinks.json
> 
> Refers to: specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/stable/2023-03-15/migrate.json:12873 in [7efa027](https://github.com/Azure/azure-rest-api-specs/commit/7efa0272dc9ada11abe239c90e8b202f5d6f2a67). [](commit_id = [7efa027](https://github.com/Azure/azure-rest-api-specs/commit/7efa0272dc9ada11abe239c90e8b202f5d6f2a67), deletion_comment = False)

Wouldn't it be better to use v5, if there's any difference?
Was there some reason we can't mix versions of common types, as long as its not the same exact 'definition' or schema?
Also ideally it'd be mandatory to refactor to common-types here since all RPs need to implement the same private links pattern.

krdhruva commented:
>         "createdTimestamp": {

Consider adding format as date-time. Pls check for all applicable instances.

---
Refers to: specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/stable/2023-03-15/migrate.json:11664 in 7efa027. [](commit_id = 7efa0272dc9ada11abe239c90e8b202f5d6f2a67, deletion_comment = False)

kuchatur-ms committed:
compile swagger with new typespec compiler version

kuchatur-ms committed:
edit

kuchatur-ms commented:
> > ```
> >     "createdTimestamp": {
> > ```
> 
> Consider adding format as date-time. Pls check for all applicable instances.
> 
> Refers to: specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/stable/2023-03-15/migrate.json:11664 in [7efa027](https://github.com/Azure/azure-rest-api-specs/commit/7efa0272dc9ada11abe239c90e8b202f5d6f2a67). [](commit_id = [7efa027](https://github.com/Azure/azure-rest-api-specs/commit/7efa0272dc9ada11abe239c90e8b202f5d6f2a67), deletion_comment = False)

We cant update this in these existing resources now. We will make sure to do this with new resources/ API URIs

kuchatur-ms committed:
Sdded PE resources using common types

kuchatur-ms committed:
added regex pattern

kuchatur-ms committed:
Merge remote-tracking branch 'origin/main' into kuchatur-ms-migrate-Microsoft.Migrate-2023-03-15-test1

kuchatur-ms committed:
edit

kuchatur-ms committed:
edit

kuchatur-ms committed:
edit

kuchatur-ms commented:
> > > ```
> > > "PrivateLinkResourceListResult": {
> > > ```
> > 
> > 
> > Consider using private link properties from common properties: https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v3/privatelinks.json
> > Refers to: specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/stable/2023-03-15/migrate.json:12873 in [7efa027](https://github.com/Azure/azure-rest-api-specs/commit/7efa0272dc9ada11abe239c90e8b202f5d6f2a67). [](commit_id = [7efa027](https://github.com/Azure/azure-rest-api-specs/commit/7efa0272dc9ada11abe239c90e8b202f5d6f2a67), deletion_comment = False)
> 
> Wouldn't it be better to use v5, if there's any difference? [Was there a reason we can't mix versions of common types, as long as its not the same exact 'definition' or schema? I don't know one.] Also ideally it'd be mandatory to refactor to common-types here since all RPs need to implement the same private links pattern.

The swagger is generated using the typespec tools. While we have updated the latest versions for the various packages, the swagger generated is still not using the v5 types. To allign with the recommended process of generating swagger (not editing manually), if we want to make this a requirement, it should be added in the tooling for typespec.
We make sure to keep our package versions updated to the latest available ones and will be able to make the changes in future

kuchatur-ms committed:
edit. minor changes from comments/ checks

kuchatur-ms committed:
cleanup

rkmanda commented:
Moving to v5 is not required esp when you are using typespec so you may ignore this error

---
In reply to: [1759178047](https://github.com/Azure/azure-rest-api-specs/pull/25417#issuecomment-1759178047) [](http://example.com/codeflow?ancestors=1759178047)

rkmanda commented:
Why is the typespec file not included in the PR? Pl refer other PRs generated by typespec and include the files. This can help simplify reviews in future as the PR reviewer can simply review the typespec file which is much simpler to review.

rkmanda commented:
>           "200": {

pl remove 200. This is also flagged here - [https://github.com/Azure/azure-rest-api-specs/pull/25417/checks?check_run_id=17638587117](https://github.com/Azure/azure-rest-api-specs/pull/25417/checks?check_run_id=17638587117)

You can achieve this thru typespec by using a custom template. Pl reach out to the typespec team for more information.  You have to declare the delete operation as ArmResourceDeleteWithoutOkAsync

---
Refers to: specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/stable/2023-03-15/migrate.json:451 in 5cabc3f. [](commit_id = 5cabc3fbce580ac8c10e4d97cd129c6d75317a6b, deletion_comment = False)

kuchatur-ms commented:
> Why is the typespec file not included in the PR? Pl refer other PRs generated by typespec and include the files. This can help simplify reviews in future as the PR reviewer can simply review the typespec file which is much simpler to review.

We have an internal repo where we are maintaining the typespec files for now. As they have been seeing some churn and might be ahead of the swagger being published with this PR, i havent added it for now. We did see some teams maintain their typespec files here too and we will try to follow that. I'll push a future PR with all typespec files added, but no functional changes to the swagger to keep it clean

kuchatur-ms committed:
made delete sync and updated examples

kuchatur-ms commented:
> > ```
> >       "200": {
> > ```
> 
> pl remove 200. This is also flagged here - https://github.com/Azure/azure-rest-api-specs/pull/25417/checks?check_run_id=17638587117
> 
> You can achieve this thru typespec by using a custom template. Pl reach out to the typespec team for more information. You have to declare the delete operation as ArmResourceDeleteWithoutOkAsync
> 
> Refers to: specification/migrate/resource-manager/Microsoft.Migrate/AssessmentProjects/stable/2023-03-15/migrate.json:451 in [5cabc3f](https://github.com/Azure/azure-rest-api-specs/commit/5cabc3fbce580ac8c10e4d97cd129c6d75317a6b). [](commit_id = [5cabc3f](https://github.com/Azure/azure-rest-api-specs/commit/5cabc3fbce580ac8c10e4d97cd129c6d75317a6b), deletion_comment = False)

The delete is supposed to return 200 and 204 only. I have updated the typespec operation to armresourcedeletesync and updated the swagger and examples.

kuchatur-ms committed:
edit. made string to date time

raych1 commented:
Approved go sdk breaking change based on the rest api bc approval.

kuchatur-ms commented:
/pr RequestMerge
